activemq-client/src/main/java/org/apache/activemq/util/JenkinsHash.java

```diff
@@ -0,0 +1,258 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.util;
+
+public class JenkinsHash {
+
+    private static long INT_MASK = 0x00000000ffffffffL;
+    private static long BYTE_MASK = 0x00000000000000ffL;
+
+    private static JenkinsHash _instance = new JenkinsHash();
```

Comment:
Would suggest adding a final there to make sure it initializes only once under multithreaded environment:

https://stackoverflow.com/questions/8865086/why-is-this-static-final-variable-in-a-singleton-thread-safe

Reply:
Thanks, we've updated the patch :)